File: heartbeat_processed/data_health_check_transformed.py
import requests
import os

source_url_health_january = "https://hadoop-and-big-data.s3-us-west-2.amazonaws.com/fitness-tracker/health_tracker_data_2020_1.json"

source_url_health_February = "https://hadoop-and-big-data.s3-us-west-2.amazonaws.com/fitness-tracker/health_tracker_data_2020_2.json"

source_url_health_February_late = "https://hadoop-and-big-data.s3-us-west-2.amazonaws.com/fitness-tracker/health_tracker_data_2020_2_late.json"

source_url_health_march   = "https://hadoop-and-big-data.s3-us-west-2.amazonaws.com/fitness-tracker/health_tracker_data_2020_3.json"

source_url_health_april = "https://hadoop-and-big-data.s3-us-west-2.amazonaws.com/fitness-tracker/health_tracker_data_2020_4.json"

source_url_health_may = "https://hadoop-and-big-data.s3-us-west-2.amazonaws.com/fitness-tracker/health_tracker_data_2020_5.json"

health_tracker_data_2020_1 = requests.get(source_url_health_january)
health_tracker_data_2020_2 = requests.get(source_url_health_February)
health_tracker_data_2020_2_late = requests.get(source_url_health_February_late)
health_tracker_data_2020_3 = requests.get(source_url_health_march)
health_tracker_data_2020_4 = requests.get(source_url_health_april)
health_tracker_data_2020_5 = requests.get(source_url_health_may)


print(health_tracker_data_2020_1.text)
print(health_tracker_data_2020_2.text)
print(health_tracker_data_2020_2_late.text)
print(health_tracker_data_2020_3.text)
print(health_tracker_data_2020_4.text)
print(health_tracker_data_2020_5.text)

path_dir_1 = r"C:\Users\lmman\OneDrive\Documents\Git\TMA Project\Azure\tma-training-DE-databricks\heartbeat_data\health_tracker_data_2020_1.txt"
path_dir_2 = r"C:\Users\lmman\OneDrive\Documents\Git\TMA Project\Azure\tma-training-DE-databricks\heartbeat_data\health_tracker_data_2020_2.txt"
path_dir_2_late = r"C:\Users\lmman\OneDrive\Documents\Git\TMA Project\Azure\tma-training-DE-databricks\heartbeat_data\health_tracker_data_2020_2_late.txt"
path_dir_3 = r"C:\Users\lmman\OneDrive\Documents\Git\TMA Project\Azure\tma-training-DE-databricks\heartbeat_data\health_tracker_data_2020_3.txt"
path_dir_4 = r"C:\Users\lmman\OneDrive\Documents\Git\TMA Project\Azure\tma-training-DE-databricks\heartbeat_data\health_tracker_data_2020_4.txt"
path_dir_5 = r"C:\Users\lmman\OneDrive\Documents\Git\TMA Project\Azure\tma-training-DE-databricks\heartbeat_data\health_tracker_data_2020_5.txt"


with open(path_dir_1,'w') as file:
    file.writelines(health_tracker_data_2020_1.text)

with open(path_dir_2,'w') as file:
    file.writelines(health_tracker_data_2020_2.text)

with open(path_dir_2_late,'w') as file:
    file.writelines(health_tracker_data_2020_2_late.text)


with open(path_dir_3,'w') as file:
    file.writelines(health_tracker_data_2020_3.text)

with open(path_dir_4,'w') as file:
    file.writelines(health_tracker_data_2020_4.text)

with open(path_dir_5,'w') as file:
    file.writelines(health_tracker_data_2020_5.text)

File: heartbeat_processed/scala/data_health_check_transformed.scala
// Databricks notebook source
import scala.io.Source

import scala.util.parsing.json._

// COMMAND ----------

val source_url_health_january: String = "https://hadoop-and-big-data.s3-us-west-2.amazonaws.com/fitness-tracker/health_tracker_data_2020_1.json"

val source_url_health_February: String = "https://hadoop-and-big-data.s3-us-west-2.amazonaws.com/fitness-tracker/health_tracker_data_2020_2.json"

val source_url_health_February_late:String = "https://hadoop-and-big-data.s3-us-west-2.amazonaws.com/fitness-tracker/health_tracker_data_2020_2_late.json"

val source_url_health_march: String = "https://hadoop-and-big-data.s3-us-west-2.amazonaws.com/fitness-tracker/health_tracker_data_2020_3.json"

 val source_url_health_april: String = "https://hadoop-and-big-data.s3-us-west-2.amazonaws.com/fitness-tracker/health_tracker_data_2020_4.json"

val source_url_health_may: String = "https://hadoop-and-big-data.s3-us-west-2.amazonaws.com/fitness-tracker/health_tracker_data_2020_5.json"


// COMMAND ----------

val health_tracker_data_2020_1 = Source.fromURL(source_url_health_january)
for (line <- health_tracker_data_2020_1.getLines) {
    println(line)
}
health_tracker_data_2020_1.close

// COMMAND ----------

val health_tracker_data_2020_2 = Source.fromURL(source_url_health_march)
for (line <- health_tracker_data_2020_2.getLines ){
  println(line)
}
health_tracker_data_2020_2.close

// COMMAND ----------

val health_tracker_data_2020_2_late = Source.fromURL(source_url_health_February_late)

for (line <- health_tracker_data_2020_2_late.getLines){
  println(line)
}
health_tracker_data_2020_2_late.close

// COMMAND ----------

val health_tracker_data_2020_3 = Source.fromURL(source_url_health_march)

for (line <-health_tracker_data_2020_3.getLines){
  println(line)
}
health_tracker_data_2020_3.close

// COMMAND ----------

val health_tracker_data_2020_4 = Source.fromURL(source_url_health_april)
for (line<- health_tracker_data_2020_4.getLines){
  println(line)
}
health_tracker_data_2020_4.close

// COMMAND ----------

val health_tracker_data_2020_5 = Source.fromURL(source_url_health_may)

for (line <- health_tracker_data_2020_5.getLines){
  val health_tracker_data_2020_5_json = line.concat(",")
  println(health_tracker_data_2020_5_json)
}
health_tracker_data_2020_5.close()

// COMMAND ----------

val json = ujson.read(Http(source_url_health_january).asString.body)
// val tuples = json("run_sessions").arr /* <-- run_sessions is an array */ .map { item => 
//     (item("id"), item("start_time"), item("end_time"), item("duration")) 
// }

// COMMAND ----------

val data = spark.read.json(health_tracker_data_2020_5)

// COMMAND ----------



// COMMAND ----------



// COMMAND ----------


